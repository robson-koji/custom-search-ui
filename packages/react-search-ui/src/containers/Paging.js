import PropTypes from "prop-types";
import RCPagination from "rc-pagination";
import React from "react";
import { withSearch } from "..";

import "rc-pagination/assets/index.css";

// App Search is currently limited to 100 pages, so we need to make sure
// that our pager only shows up to 100 pages.
function limitedTo100Pages(totalResults, resultsPerPage) {
  return Math.min(resultsPerPage * 100, totalResults);
}

export function PagingContainer({
  current,
  resultsPerPage,
  setCurrent,
  totalResults
}) {
  if (totalResults === 0) return null;

  return (
    <div>
      <RCPagination
        pageSize={resultsPerPage}
        current={current}
        total={limitedTo100Pages(totalResults, resultsPerPage)}
        onChange={setCurrent}
      />
    </div>
  );
}

PagingContainer.propTypes = {
  current: PropTypes.number.isRequired,
  resultsPerPage: PropTypes.number.isRequired,
  setCurrent: PropTypes.func.isRequired,
  totalResults: PropTypes.number.isRequired
};

export default withSearch(PagingContainer);
