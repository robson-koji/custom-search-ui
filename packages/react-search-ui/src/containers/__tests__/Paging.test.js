import React from "react";
import { shallow } from "enzyme";
import { PagingContainer } from "../Paging";

const params = {
  current: 1,
  resultsPerPage: 20,
  setCurrent: jest.fn(),
  totalResults: 100
};

beforeEach(() => {
  params.setCurrent = jest.fn();
});

it("renders correctly", () => {
  const wrapper = shallow(<PagingContainer {...params} />);
  expect(wrapper).toMatchSnapshot();
});

it("limits total results if they would result in over 100 pages", () => {
  const wrapper = shallow(
    <PagingContainer
      {...{
        ...params,
        resultsPerPage: 10,
        totalResults: 398183013080
      }}
    />
  );

  expect(wrapper.find("Pagination").prop("total")).toEqual(1000);
});

it("renders empty when there are no results", () => {
  const wrapper = shallow(
    <PagingContainer
      {...{
        ...params,
        totalResults: 0
      }}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it("will call back when a the page is changed", () => {
  const wrapper = shallow(<PagingContainer {...params} />);

  wrapper.find("Pagination").prop("onChange")(2);

  const current = params.setCurrent.mock.calls[0][0];
  expect(current).toEqual(2);
});
