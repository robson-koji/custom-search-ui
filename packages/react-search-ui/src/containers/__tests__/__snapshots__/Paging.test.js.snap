// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PagingContainer
    current={1}
    resultsPerPage={20}
    setCurrent={[MockFunction]}
    totalResults={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Pagination
        className=""
        current={1}
        defaultCurrent={1}
        defaultPageSize={10}
        hideOnSinglePage={false}
        itemRender={[Function]}
        locale={
          Object {
            "items_per_page": "条/页",
            "jump_to": "跳至",
            "jump_to_confirm": "确定",
            "next_3": "向后 3 页",
            "next_5": "向后 5 页",
            "next_page": "下一页",
            "page": "页",
            "prev_3": "向前 3 页",
            "prev_5": "向前 5 页",
            "prev_page": "上一页",
          }
        }
        onChange={[MockFunction]}
        onShowSizeChange={[Function]}
        pageSize={20}
        prefixCls="rc-pagination"
        selectComponentClass={null}
        selectPrefixCls="rc-select"
        showLessItems={false}
        showPrevNextJumpers={true}
        showQuickJumper={false}
        showSizeChanger={false}
        showTitle={true}
        style={Object {}}
        total={100}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "",
        "current": 1,
        "defaultCurrent": 1,
        "defaultPageSize": 10,
        "hideOnSinglePage": false,
        "itemRender": [Function],
        "locale": Object {
          "items_per_page": "条/页",
          "jump_to": "跳至",
          "jump_to_confirm": "确定",
          "next_3": "向后 3 页",
          "next_5": "向后 5 页",
          "next_page": "下一页",
          "page": "页",
          "prev_3": "向前 3 页",
          "prev_5": "向前 5 页",
          "prev_page": "上一页",
        },
        "onChange": [MockFunction],
        "onShowSizeChange": [Function],
        "pageSize": 20,
        "prefixCls": "rc-pagination",
        "selectComponentClass": null,
        "selectPrefixCls": "rc-select",
        "showLessItems": false,
        "showPrevNextJumpers": true,
        "showQuickJumper": false,
        "showSizeChanger": false,
        "showTitle": true,
        "style": Object {},
        "total": 100,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Pagination
          className=""
          current={1}
          defaultCurrent={1}
          defaultPageSize={10}
          hideOnSinglePage={false}
          itemRender={[Function]}
          locale={
            Object {
              "items_per_page": "条/页",
              "jump_to": "跳至",
              "jump_to_confirm": "确定",
              "next_3": "向后 3 页",
              "next_5": "向后 5 页",
              "next_page": "下一页",
              "page": "页",
              "prev_3": "向前 3 页",
              "prev_5": "向前 5 页",
              "prev_page": "上一页",
            }
          }
          onChange={[MockFunction]}
          onShowSizeChange={[Function]}
          pageSize={20}
          prefixCls="rc-pagination"
          selectComponentClass={null}
          selectPrefixCls="rc-select"
          showLessItems={false}
          showPrevNextJumpers={true}
          showQuickJumper={false}
          showSizeChanger={false}
          showTitle={true}
          style={Object {}}
          total={100}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "",
          "current": 1,
          "defaultCurrent": 1,
          "defaultPageSize": 10,
          "hideOnSinglePage": false,
          "itemRender": [Function],
          "locale": Object {
            "items_per_page": "条/页",
            "jump_to": "跳至",
            "jump_to_confirm": "确定",
            "next_3": "向后 3 页",
            "next_5": "向后 5 页",
            "next_page": "下一页",
            "page": "页",
            "prev_3": "向前 3 页",
            "prev_5": "向前 5 页",
            "prev_page": "上一页",
          },
          "onChange": [MockFunction],
          "onShowSizeChange": [Function],
          "pageSize": 20,
          "prefixCls": "rc-pagination",
          "selectComponentClass": null,
          "selectPrefixCls": "rc-select",
          "showLessItems": false,
          "showPrevNextJumpers": true,
          "showQuickJumper": false,
          "showSizeChanger": false,
          "showTitle": true,
          "style": Object {},
          "total": 100,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders empty when there are no results 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PagingContainer
    current={1}
    resultsPerPage={20}
    setCurrent={[MockFunction]}
    totalResults={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
